errorlint/lint.go

```diff
@@ -13,6 +13,139 @@ import (
 	"golang.org/x/tools/go/analysis"
 )
 
+type diagnosticType int
+
+const (
+	typeAssertionDiag diagnosticType = iota
+	errorComparisonDiag
+	otherDiag
+)
+
+const (
+	typeAssertionPattern   = "type assertion on error"
+	errorComparisonPattern = "comparing with"
+)
+
+// classifyDiagnostic determines the type of diagnostic based on its message
+func classifyDiagnostic(diagnostic analysis.Diagnostic) diagnosticType {
+	msg := diagnostic.Message
+	if strings.Contains(msg, typeAssertionPattern) {
+		return typeAssertionDiag
+	}
+	if strings.Contains(msg, errorComparisonPattern) {
+		return errorComparisonDiag
+	}
+	return otherDiag
+}
```

Comment:
I'm surprised the detection is made on the message.

Should we use this constant when raising the diagnostic? Maybe I missed something 🤔